rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -105,11 +119,31 @@ Recorder::get_missing_topics(const std::unordered_map<std::string, std::string>
   return missing_topics;
 }
 
+namespace
+{
+std::string serialized_offered_qos_profiles_for_topic(
+  std::shared_ptr<rosbag2_transport::Rosbag2Node> node,
+  const std::string & topic_name)
+{
+  YAML::Node offered_qos_profiles;
+  auto publishers_info = node->get_publishers_info_by_topic(topic_name);
+  for (auto info : publishers_info) {
+    offered_qos_profiles.push_back(rosbag2_transport::Rosbag2QoS(info.qos_profile()));
+  }
+  return YAML::Dump(offered_qos_profiles);
+}
+}  // unnamed namespace
```

Comment:
Does the linter accept this? I thought for anonymous namespaces it wanted `// namespace`

Reply:
Note: I learned recently that the technical term is "unnamed namespace" and that "anonymous namespace" isn't a concept in the C++ spec. https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces. I've been calling them anonymous for years lol.

But, yes, the linter accepts it

Reply:
TIL: I've been calling it the wrong name for a whole decade

Reply:
Maybe an old version of the spec used that terminology, idk. Maybe it's the mandela effect

Reply:
https://docs.microsoft.com/en-us/cpp/cpp/namespaces-cpp?view=vs-2019#anonymous-or-unnamed-namespaces